iceoryx_utils/test/moduletests/test_posix_timer.cpp

```diff
@@ -65,40 +65,65 @@ TEST_F(TimerStopWatch_test, DurationOfZeroCausesError)
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_TRUE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_TRUE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetWithDurationIsExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
-    TIMING_TEST_EXPECT_TRUE(sut.hasExpiredComparedToCreationTime());
-    sut.resetCreationTime();
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_TRUE(sut.hasExpired());
+    sut.reset();
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetWhenNotExpiredIsStillNotExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    sut.resetCreationTime();
+    sut.reset();
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetAfterBeingExpiredIsNotExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
 
-    TIMING_TEST_ASSERT_TRUE(sut.hasExpiredComparedToCreationTime());
-    sut.resetCreationTime();
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_ASSERT_TRUE(sut.hasExpired());
+    sut.reset();
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
+TIMING_TEST_F(TimerStopWatch_test, ResetWithCustomizedTimeAfterBeingExpiredIsNotExpired, Repeat(5), [&] {
+    iox::clock::DeadlineTimer sut(TIMEOUT);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+
+    TIMING_TEST_ASSERT_TRUE(sut.hasExpired());
+
+    sut.reset(20_s);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
+});
+
+
+TIMING_TEST_F(TimerStopWatch_test, RemainingTimeCheckIfNotExpired, Repeat(5), [&] {
+    iox::clock::DeadlineTimer sut(TIMEOUT);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+
+    TIMING_TEST_ASSERT_TRUE(sut.hasExpired());
+
+    sut.reset(20_s);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+    EXPECT_THAT(sut.remainingTime(), Le(TIMEOUT));
```

Comment:
This test will not work because you reset the timer with 20 Seconds and expect then a value lower or equal to 20 milliseconds after waiting 40 milliseconds. The logic is also incorrect because the test would also pass if no time has passed (Lower equal)

Proposal:
1. create a timer and wait for a defined time
2. get the remaining time
3. check if the result is in a range ±10 percent from the expected value

You need to make a range because the test could flicker due to the non-deterministic Github CI.
For an example please take a look at `TIMING_TEST_F(Timer_test, RestartWithDifferentTiming)` in line 237 of this file.

Reply:
I am stuck in fixing this test case. I just found that the test cases are not executing for me locally. I am using the script to build `./tools/iceoryx_build_test.sh clean build-test build-strict test debug`. However the posix timer tests are not available in the posh_moduletests executable inside the build folder. Pls suggest me how i should enable the posix timer test to run locally.